### ISSUE TITLE
TableReport doesn't like python objects like nested sklearn estimators

### ISSUE THREAD

**Issue by Vincent-Maladiere** [skrub-data/skrub#1195]
### Describe the bug

In the example below, `TableReport` uses the string representation of the sklearn pipeline as a category.

<img width="749" alt="Screenshot 2024-12-11 at 16 54 33" src="https://github.com/user-attachments/assets/b097b72c-3644-4800-ae59-cb4fa1e2bd32" />

### Steps/Code to Reproduce

```python
from skrub import TableVectorizer, tabular_learner

TableReport(
    pd.DataFrame(
        dict(
            a=[
                tabular_learner("regressor"),
                tabular_learner("classifier"),
            ]
        )
    )
)
```

### Expected Results

A nonnested representation of the object? In the worst-case scenario, its Python id? Not sure

### Actual Results

See screenshot above

### Versions

```shell
'0.4.dev0'
```

**Comment by jeromedockes** [comment]
thanks! we need to cast to string before processing, truncating and plotting.

another reproducer

```python
import skrub
import pandas as pd

class A:
    def __repr__(self):
        return """
        some


        big



        repr

        """

df = pd.DataFrame({"a": [A(), A()]})

skrub.TableReport(df).open()

```

**Comment by jeromedockes** [comment]
let's fix it before doing 0.4.1
